solutions/community-deploy/communitybuild.sh

```diff
@@ -0,0 +1,141 @@
+        fi
+}
+docker_check
+
+function docker_compose_check() {
+        docker-compose --version >/dev/null 2>&1
+        if [ $? -eq 0 ]; then
+                add_common_separator "Installed Docker-Compose version: $(docker-compose --version)."
+        else
+                add_common_separator "Docker-compose not installed"
+                add_primary_separator "Installing Docker-compose"
+                curl -SL https://github.com/docker/compose/releases/download/v2.5.0/docker-compose-linux-x86_64 -o /usr/local/bin/docker-compose
+                chmod +x /usr/local/bin/docker-compose
+                ln -s /usr/local/bin/docker-compose /usr/bin/docker-compose
+                docker-compose --version
+        fi
+}
+docker_compose_check
+
+#Install git
+yum install git -y
+
+# Compile and Build CORTX Stack
+docker rmi -f $(docker images | grep -i 'ghcr.io/seagate/cortx-build*')
+docker pull ghcr.io/seagate/cortx-build:rockylinux-8.4
+
+# Clone the CORTX repository
+pushd /mnt && git clone https://github.com/Seagate/cortx --recursive --depth=1
+
+# Checkout main branch for generating CORTX packages
+docker run --rm -v /mnt/cortx:/cortx-workspace ghcr.io/seagate/cortx-build:rockylinux-8.4 make checkout BRANCH=$BRANCH
+
+# Validate CORTX component clone status
+pushd /mnt/cortx/ 
+for component in cortx-motr cortx-hare cortx-rgw-integration cortx-manager cortx-utils cortx-ha cortx-rgw
+do 
+echo -e "\n==[ Checking git branch for $component ]=="
+pushd $component
+git status | egrep -iw 'Head|modified|Untracked'
+    if [ $? -eq 0 ]; then
+       add_common_separator "Git status pending"
+        exit 1
+    else    
+        popd
+    fi    
+done && pushd -
+
+# Build the CORTX packages
+docker run --rm -v /var/artifacts:/var/artifacts -v /mnt/cortx:/cortx-workspace ghcr.io/seagate/cortx-build:rockylinux-8.4 make clean cortx-all-rockylinux-image
+
+function packet_validation() {
+        ls -l /var/artifacts/0 | egrep -iw '3rd_party|python_deps'
+        if [ $? -eq 0 ]; then
+            add_common_separator "Required packages are available"
+        else
+            add_common_separator "Required packages are not available to proceed further"
+            exit 1
+        fi
+}
+packet_validation
+
+# Nginx container creation with required configuration
+docker run --name release-packages-server -v /var/artifacts/0/:/usr/share/nginx/html:ro -d -p 80:80 nginx
+
+function nginx_validation() {
+        sleep 60
+        docker ps | grep -iw 'nginx'
+        if [ $? -eq 0 ]; then
+            curl -L http://$IP/RELEASE.INFO
+        else
+            exit 1
+}
+nginx_validation
+# clone cortx-re repository & run build.sh
+git clone https://github.com/Seagate/cortx-re && pushd cortx-re/docker/cortx-deploy/
+./build.sh -b http://$IP -o rockylinux-8.4 -s all -e opensource-ci
+
+# Show recently generated cortx-all images
+docker images --format='{{.Repository}}:{{.Tag}} {{.CreatedAt}}' --filter=reference='cortx-*'
+
+# Execution
```

Comment:
Please execute all functions at the end.

Reply:
As discussed with Shailesh, we will check on function declaration place later.